### PR TITLE
build: :hammer: add targets pipeline to rebuild data files

### LINKED ISSUE #81: Implement a targets pipeline to rebuild `data/` objects whenever something changes in the scripts

### PR DESCRIPTION
**Pull request by lwjohnst86**
Closes #81

Use `targets::tar_make()` to rebuild the data files whenever there are changes to `data-raw/` files.

### CODE REVIEW

File: data-raw/algorithm.R [hunk 1]
```diff
@@ -0,0 +1,5 @@
+create_algorithm_data <- function(path) {
+  algorithm <- readr::read_csv(path, show_col_types = FALSE)
+  usethis::use_data(algorithm, overwrite = TRUE)
+  fs::dir_ls(here::here("data"), regexp = "algorithm")
```
Comment (lwjohnst86) [inline data-raw/algorithm.R:4]: Functions as part of a targets pipeline must output the file path if the function code saves to a file.

File: _targets/meta/meta [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+variable_description_rda|stem|450aefe34ecbfb32|1742010767fda041|824677c99eb59944|-1262878967|/home/luke/Documents/organizations/steno-aarhus/osdc/data/variable_description.rda|t19893.4447566339s|5272409a13fc9df9|1550|file|local|vector|||0.496|Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.ℹ Please use register_name instead of .dataregister_name. Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.ℹ Please use register_abbrev instead of .dataregister_abbrev. Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.ℹ Please use variable_name instead of .datavariable_name. Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.ℹ Please use start_year instead of .datastart_year. Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.ℹ Please use end_year instead of .dataend_year. Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.ℹ Please use danish_description instead of .datadanish_description. Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.ℹ Please use english_description instead of .dataenglish_description|
```
Comment (Aastedet) [inline _targets/meta/meta:16]: This tidyselect warning is ugly, and seems simple to address, if the warning message is accurate. Should we create an issue about it?

**Review by Aastedet — APPROVED** [review]

File: _targets/meta/meta (hunk 2 above)
Comment (lwjohnst86) [reply _targets/meta/meta:16]: Yes, this is fixed in #127 ☺️

File: DESCRIPTION
Comment (signekb) [inline DESCRIPTION:1]: should targets be added to the DESCRIPTION file?

File: _targets.R [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      usethis::use_data(variable_description, overwrite = TRUE)
+      here::here("data/variable_description.rda")
+    },
+    format = "file"
+  ),
+  tar_target(
+    name = internal_rda,
+    command = {
+      usethis::use_data(algorithm, variable_description, overwrite = TRUE, internal = TRUE)
+      here::here("R/sysdata.rda")
```
Comment (signekb) [inline _targets.R:60]: Should `sysdata.rda` be sourced like algorithm and variable_description at L16-18? Or is that not necessary bc it's just overwritten here?

Just thinking about this bc I got this message when running `targets::tar_make()`: 
> tar_source() only sources R scripts. Ignoring non-R files: R/sysdata.rda

File: _targets.R [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+# Replace the target list below with your own:
+list(
+  tar_target(
+    name = algorithm_csv,
+    command = "data-raw/algorithm.csv",
+    format = "file"
+  ),
+  tar_target(
+    name = algorithm,
```
Comment (signekb) [inline _targets.R:28]:
```suggestion
    name = algorithm_data,
```
To align naming with the function `read_algorithm_data` below

File: _targets.R [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    command = read_algorithm_data(algorithm_csv)
+  ),
+  tar_target(
+    name = algorithm_rda,
+    command = {
+      usethis::use_data(algorithm, overwrite = TRUE)
```
Comment (signekb) [inline _targets.R:34]:
```suggestion
      usethis::use_data(algorithm_data, overwrite = TRUE)
```

File: _targets.R [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    format = "file"
+  ),
+  tar_target(
+    name = variable_description_csv,
+    command = "data-raw/variable-description.csv",
+    format = "file"
+  ),
+  tar_target(
+    name = variable_description,
```
Comment (signekb) [inline _targets.R:45]:
```suggestion
    name = variable_description_data,
```

File: _targets.R [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    command = read_variable_description_data(variable_description_csv)
+  ),
+  tar_target(
+    name = variable_description_rda,
+    command = {
+      usethis::use_data(variable_description, overwrite = TRUE)
```
Comment (signekb) [inline _targets.R:51]:
```suggestion
      usethis::use_data(variable_description_data, overwrite = TRUE)
```

File: README.md [hunk 8]
```diff
@@ -41,10 +41,20 @@ more data or running the full test suite, use:
 ``` r
 pak::pak(
   "steno-aarhus/osdc",
-  dependencies = c("all", "Config/Needs/tests", "Config/Needs/data")
+  dependencies = c("all", "Config/Needs/data")
```
Comment (signekb) [inline README.md:44]: Why did you remove the tests here? 
On L39 above it says "[...] or running the full test suite, use:". Is the `tests` not needed for that?

**Review by signekb — COMMENTED** [review]
I think this works as expected 👍  
I just have a few questions and a suggestion for renaming a few targets.

File: README.md (hunk 8 above)
Comment (lwjohnst86) [reply README.md:44]: There's a CRAN check issue that I am trying to debug and I want to test if this will fix it. It might be brought back depending on the issue.

File: _targets.R (hunk 4 above)
Comment (lwjohnst86) [reply _targets.R:28]: The name here would be the name that we use internally to access those objects (e.g. with using `use_data()`. It is a bit redundant to add `_data` to the end of the object when it is a data frame already (it would be a bit like calling all variables that are strings with `_string` like `age_string` or `drugs_string`).

File: _targets.R (hunk 3 above)
Comment (lwjohnst86) [reply _targets.R:60]: Those are two separate things. The message comes because there is the `R/sysdata.rda` file in the `R/` folder. And using `tar_source()` only loads R files. So it is only a message. We don't source the `sysdata.rda` in line 16 because we don't need it for targets. `sysdata.rda` contains objects used within functions of the package, that are internal to the package.
